Sub task benchmark

chc012 opened the pull request.

sanjari-orb commented on browsergym/subtaskbench/src/browsergym/subtaskbench/task.py:
```diff
@@ -0,0 +1,146 @@
+        from subtaskbench import Evaluator
+
+        self.evaluator = Evaluator(
+            start_url=self.task_config["start_url"],
```
Hmm so the start URL field might be a problem for this benchmark, because it will be ephemeral and thus cannot be read from a pre-defined config. One thing we can do let this be an environment id and pass it to a an handler in this file which can return the correct start url (can be left empty for now).
The same applies to the creation of the `playwright.sync_api.Page` as well.

chc012 replied:
So I do not envision this GenericSubTaskBenchTask class to be used by itself (maybe I should mark it abstract). Instead I have the Static....task and the online...task down below as its subclasses. in the online tasks setup, you'll see that i have a special logic where i take in a env var called SUBTASKBENCH_ENDPOINT, and use that to replace all occurrence of "__ENDPOINT__" in the start url of a served env. I think that is what you are referring to here?

sanjari-orb replied:
I see.. ok we can let it be as it is in that case.

sanjari-orb commented on browsergym/subtaskbench/src/browsergym/subtaskbench/task.py:
```diff
@@ -0,0 +1,146 @@
+            start_url=self.task_config["start_url"],
+            goal=self.task_config["goal"],
+            evaluation_script=self.task_config["evaluation_script"],
+            expected_output=self.task_config["expected_output"],
```
What is the expected output field expected to look like? Is this for information seeking tasks? (eg, string_match, fuzzy match type evals)

chc012 replied:
Yes, so we can specify what we want the agent to output in case needed. exactly how this is going to be used is up to the us (the imagined subtaskbench library).

chc012 commented:
I addressed some concerns here. I will not mark it as completed or merge until we have the subtaskbench library figured out (collecting and serving warc, static website, etc.) fyi @sanjari-orb

chc012 commented on browsergym/visualwebarena/src/browsergym/visualwebarena/task.py:
```diff
@@ -144,7 +144,7 @@ def __init__(
 
```
Oh 🤣 reverting this change actually broke unit testing cuz they have linter enforced. really makes me wonder how they merged these files in in the first place...

sanjari-orb commented:
feel free to merge as long as it does not break the existing classes
